File: FRONTEND/src/app/objects/borne.ts
export class Borne{
    constructor(
        public id: number,
        public nomEnseigne: string,
        public station: Station,
        public priseType: Prise[],
        public horaires: string,
        public miseEnService: Date | null,
        public coordonnees: Coordonnees,
        public ville: Ville,
    ){}
}
export class BornePoint{
  constructor(
    public id: number,
    public latitude: number,
    public longitude: number,
    public date: Date | null
  ) {
  }
}
export class Station{
    constructor(
        public id: number,
        public nom: string,
        public address: string
    ){}
}
export class Prise{
  constructor(
    public id: number,
    public name: string,
    public code: string
  ) {
  }
}
export class Coordonnees {
    constructor(
        public id: number,
        public longitude: number,
        public latitude: number
    ){}
}
export class Ville {
    constructor(
      public id: number,
      public commune: string,
      public code_postale: string
    ){}
}

File: DBobjects/src/main/kotlin/fr/esir/vehicules/dbobjects/bornes/Borne.kt
package fr.esir.vehicules.dbobjects.bornes

import jakarta.persistence.CascadeType
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.JoinTable
import jakarta.persistence.ManyToMany
import jakarta.persistence.ManyToOne
import jakarta.persistence.OneToOne
import java.sql.Date

@Entity
class Borne (
        @Id
        @GeneratedValue
        val id: Int?=null,

        @Column
        val nomEnseigne: String,
        @OneToOne(cascade = [CascadeType.ALL])
        @JoinColumn
        val station: Station,
        @ManyToMany(cascade = [CascadeType.DETACH])
        @JoinTable
        val priseType: Set<Prise>,
        @Column
        val horaires: String,
        @Column
        val miseEnService: Date ?,
        @OneToOne(cascade = [CascadeType.ALL])
        @JoinColumn
        val coordonnees: Coordonnees,
        @ManyToOne(cascade = [CascadeType.ALL])
        @JoinColumn
        val ville: Ville
)